src/main/java/appeng/core/sync/packets/PacketTransitionEffect.java

```diff
@@ -20,45 +20,37 @@
 
 
 import io.netty.buffer.ByteBuf;
-import io.netty.buffer.Unpooled;
 
 import net.minecraft.block.Block;
 import net.minecraft.client.Minecraft;
 import net.minecraft.client.audio.PositionedSoundRecord;
-import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.init.Items;
 import net.minecraft.util.ResourceLocation;
 import net.minecraft.world.World;
 import net.minecraftforge.common.util.ForgeDirection;
 
-import cpw.mods.fml.relauncher.Side;
-import cpw.mods.fml.relauncher.SideOnly;
+import cpw.mods.fml.common.network.simpleimpl.MessageContext;
 
 import appeng.client.ClientHelper;
 import appeng.client.render.effects.EnergyFx;
 import appeng.core.CommonHelper;
 import appeng.core.sync.AppEngPacket;
-import appeng.core.sync.network.INetworkInfo;
+import appeng.core.sync.AppEngPacketHandler;
 import appeng.util.Platform;
 
 
-public class PacketTransitionEffect extends AppEngPacket
+public class PacketTransitionEffect implements AppEngPacket, AppEngPacketHandler<PacketTransitionEffect, AppEngPacket>
 {
 
-	public final boolean mode;
-	final double x;
-	final double y;
-	final double z;
-	final ForgeDirection d;
+	private boolean mode;
+	private double x;
+	private double y;
+	private double z;
+	private ForgeDirection d;
 
 	// automatic.
-	public PacketTransitionEffect( ByteBuf stream )
+	public PacketTransitionEffect()
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1186)